src/TorchSharp/NN/Module.cs

```diff
@@ -142,12 +143,74 @@ protected virtual void Dispose(bool disposing)
                     }
                 }
 
+                [DllImport("LibTorchSharp")]
+                static extern void THSNN_Module_to_device_dtype(HType module, sbyte dtype, long deviceType, long deviceIndex);
+
                 [DllImport("LibTorchSharp")]
                 static extern void THSNN_Module_to_device(HType module, long deviceType, long deviceIndex);
 
                 [DllImport("LibTorchSharp")]
                 static extern void THSNN_Module_to_dtype(HType module, sbyte dtype);
 
+                /// <summary>
+                /// Moves and converts the parameters and buffers.
+                /// </summary>
+                /// <param name="device">The target device.</param>
+                /// <param name="dtype">The target element type.</param>
+                public virtual Module to(Device device, ScalarType dtype)
+                {
+                    if (device.type != DeviceType.CUDA) { device = new Device(device.type, -1); };
+
+                    if (device.type == DeviceType.CUDA && !torch.cuda.is_available()) throw new InvalidOperationException("CUDA is not available.");
+
+                    InitializeDeviceType(device.type);
+                    THSNN_Module_to_device_dtype(handle, (sbyte)dtype, (int)device.type, device.index);
+                    CheckForErrors();
+
+                    _toEpilog(device, dtype);
+
+                    return this;
+                }
+
+                protected void _toEpilog(Device device, ScalarType dtype)
```

Comment:
>_toEpilog [](http://example.com/codeflow?start=31&length=9)

The code here is identical to the code in the same method that takes DeviceType. Is there a way we can share the code instead of duplicating it? Or we can make this method a generic method?